pkg/admission/handlers/sts.go

```diff
@@ -0,0 +1,57 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/kanopy-platform/hedgetrimmer/pkg/admission"
+	"github.com/kanopy-platform/hedgetrimmer/pkg/limitrange"
+	appsv1 "k8s.io/api/apps/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	kadmission "sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+type STSHandler struct {
+	DefaultDecoderInjector
+	ptm admission.PodTemplateSpecMutator
+}
+
+func NewSTSHandler(ptm admission.PodTemplateSpecMutator) *STSHandler {
+	return &STSHandler{ptm: ptm}
+}
+
+func (sts *STSHandler) Kind() string { return "StatefulSet" }
+
+func (sts *STSHandler) Handle(ctx context.Context, req kadmission.Request) kadmission.Response {
+	log := log.FromContext(ctx)
+
+	lrConfig, ok := ctx.Value(limitrange.LimitRangeContextTypeMemory).(*limitrange.Config)
+	if !ok || lrConfig == nil {
+		reason := fmt.Sprintf("failed to list LimitRanges in namespace: %s", req.Namespace)
+		log.Error(fmt.Errorf(reason), reason)
+		//If we cannot get LimitRanges due to an api error fail.
+		return kadmission.Denied(reason)
+	}
```

Comment:
This seems like it should also be a helper method.  Is a check necessary here, It's only used in the call to `ptm.Mutate` which will error if `lrConfig` is nil or invalid.

Reply:
Mutate doesn't expect a nil and will probably panic vs error. 

This could be a helper but similar to PatchResponse.

Reply:
It doesn't make it much more concise for logging but it does internalize the context type which I like. I can add a similar helper for the handlers so neither needs to know about the wrapped types.